addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -682,6 +684,21 @@ function factory(dependencies) {
             });
         }
 
+        /**
+         * @private
+         * @param {integer} channelId
+         */
+        _handleNotificationPartnerToggleMassMailing(channelId) {
+            const channel = this.env.models['mail.thread'].findFromIdentifyingData({
+                id: channelId,
+                model: 'mail.channel',
+            });
+            if (!channel) {
+                return;
+            }
+            channel.update({ mass_mailing: !channel.mass_mailing });
```

Comment:
According to other suggestion, this should receive `mass_mailing` as param and simply update to new value.